make/common/NativeCompilation.gmk

```diff
@@ -221,6 +221,26 @@ define SetupNativeCompilationBody
     $1 := $$($1_ALL_OBJS_JSON) $$($1_LDFLAGS_FILE)
     TARGETS := $(filter $(MAKESUPPORT_OUTPUTDIR)/compile-commands/%, $(TARGETS))
   endif
+
+  # Call signing hook
+  $$(eval $$(call CallSigningHook,$1))
+
+endef
+
+################################################################################
+define CallSigningHook
+  # Create a rule to call post-link signing hook
+  $1_SIGNING_OUTPUT_PREFIX := $$($1_OBJECT_DIR)/$$($1_SAFE_NAME)_call_signing_hook
+  $1_CALL_SIGNING_HOOK := $$($1_SIGNING_OUTPUT_PREFIX).log
+
+  $$($1_CALL_SIGNING_HOOK): $$($1_TARGET)
+        ifeq ($(SIGNING_HOOK_ENABLED), true)
+	  $$(call LogInfo, Signing $$($1_BASENAME))
+	  $$(call ExecuteWithLog, $$($1_SIGNING_OUTPUT_PREFIX), \
+	      $(FIXPATH) $(SIGNING_HOOK) $$($1_TARGET))
+        endif
+
+  $1 += $$($1_CALL_SIGNING_HOOK)
 endef
```

Comment:
You can simplify this, so you get rid of `SIGNING_HOOK_ENABLED`. Also, FIXPATH is not needed after the call to `UTIL_FIXUP_EXECUTABLE`.

```suggestion
define CallSigningHook
  ifneq ($(SIGNING_HOOK), )
    # Create a rule to call post-link signing hook
    $1_SIGNING_OUTPUT_PREFIX := $$($1_OBJECT_DIR)/$$($1_SAFE_NAME)_call_signing_hook
    $1_CALL_SIGNING_HOOK := $$($1_SIGNING_OUTPUT_PREFIX).log

    $$($1_CALL_SIGNING_HOOK): $$($1_TARGET)
	$$(call LogInfo, Signing $$($1_BASENAME))
	$$(call ExecuteWithLog, $$($1_SIGNING_OUTPUT_PREFIX), \
	    $(SIGNING_HOOK) $$($1_TARGET))

    $1 += $$($1_CALL_SIGNING_HOOK)
  endif
endef
```